app/components/starred-docs.ts

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+export default function(app) {
+    app.component('starredDocs', {
+      bindings: {
+        personId: '<',
+      },
+      controller: [
+        '$scope', '$element', '$attrs', '$http', 'config', 'notificationService',
+        function($scope, $element, $attrs, $http, config, notificationService) {
+          const ctrl = this;
+
+          $scope.$watch('$ctrl.personId', async function(id) {
+            if (!id) { return; }
+            let ret;
+            try {
+              ret = await $http.get(
+                config.apiUrl +
+                  '/people/' + id + '/stars'
+              );
+            } catch (err) {
+              notificationService.notifications.push({
+                type: 'error',
+                message: err.data.message ? err.data.message :
+                  'could not fetch starred documents (unknown reason)'
+              });
+            }
+            ctrl.starredDocuments = ret.data.documents;
```

Comment:
shouldn't this be inside the `try` block? Then `ret` could also move there and be declared as `const`...